cpp/sopt/wavelets/innards.impl.h

```diff
@@ -5,8 +5,7 @@
 #include <Eigen/Core>
 
 // Function inside anonymouns namespace won't appear in library
-namespace sopt {
-namespace wavelets {
+namespace sopt::wavelets {
 namespace {
```

Comment:
Why do we have this unnamed namespace as well? It seems to contain all the tings that are in `spot::wavelets`, does it serve any function?

Reply:
Ahh I've just seen the comment above that explains why they did this. Is this a good practice?

Reply:
Yes. It is a reasonable practice. As per learncpp, https://www.learncpp.com/cpp-tutorial/unnamed-and-inline-namespaces/

>The other effect of unnamed namespaces is that all identifiers inside an unnamed namespace are treated as if they had internal linkage, which means that the content of an unnamed namespace can’t be seen outside of the file in which the unnamed namespace is defined.
> Unnamed namespaces are typically used when you have a lot of content that you want to ensure stays local to a given file, as it’s easier to cluster such content in an unnamed namespace than individually mark all declarations as static. Unnamed namespaces will also keep user-defined types (something we’ll discuss in a later lesson) local to the file, something for which there is no alternative equivalent mechanism to do.